src/lib/services/transfer-service/TransferService.ts

```diff
@@ -1,27 +1,233 @@
+import { RegistryServiceApi } from 'lib/api/registry-service-api/registryServiceApi';
+import { DiscoveryServiceApi } from 'lib/api/discovery-service-api/discoveryServiceApi';
+import { SubmodelRegistryServiceApi } from 'lib/api/submodel-registry-service/submodelRegistryServiceApi';
+import { AssetAdministrationShellDescriptor, Endpoint, SubmodelDescriptor } from 'lib/types/registryServiceTypes';
+import { TransferDto, TransferResult } from 'lib/services/transfer-service/transferActions';
+import { encodeBase64 } from 'lib/util/Base64Util';
+import { AssetAdministrationShell, Submodel } from '@aas-core-works/aas-core3.0-typescript/types';
 
 export class TransferService {
     private constructor(
         protected readonly targetAasRepositoryClient: IAssetAdministrationShellRepositoryApi,
-        protected readonly targetSubmodelRepositoryClient: SubmodelRepositoryApi,
+        protected readonly targetSubmodelRepositoryClient: ISubmodelRepositoryApi,
+        protected readonly targetAasDiscoveryClient?: IDiscoveryServiceApi,
+        protected readonly targetAasRegistryClient?: IRegistryServiceApi,
+        protected readonly targetSubmodelRegistryClient?: ISubmodelRegistryServiceApiInterface,
     ) {}
 
-    static create(targetAasRepositoryBaseUrl: string, targetSubmodelRepositoryBaseUrl: string): TransferService {
+    static create(
+        targetAasRepositoryBaseUrl: string,
+        targetSubmodelRepositoryBaseUrl: string,
+        targetAasDiscoveryBaseUrl?: string,
+        targetAasRegistryBaseUrl?: string,
+        targetSubmodelRegistryBaseUrl?: string,
+    ): TransferService {
         const targetAasRepositoryClient = AssetAdministrationShellRepositoryApi.create({
             basePath: targetAasRepositoryBaseUrl,
             fetch: mnestixFetch(),
         });
+
         const targetSubmodelRepositoryClient = SubmodelRepositoryApi.create({
             basePath: targetSubmodelRepositoryBaseUrl,
             fetch: mnestixFetch(),
         });
-        return new TransferService(targetAasRepositoryClient, targetSubmodelRepositoryClient);
+
+        const targetAasDiscoveryClient = targetAasDiscoveryBaseUrl
+            ? DiscoveryServiceApi.create(targetAasDiscoveryBaseUrl, mnestixFetch())
+            : undefined;
+
+        const targetAasRegistryClient = targetAasRegistryBaseUrl
+            ? RegistryServiceApi.create(targetAasRegistryBaseUrl, mnestixFetch())
+            : undefined;
+
+        const targetSubmodelRegistryClient = targetSubmodelRegistryBaseUrl
+            ? SubmodelRegistryServiceApi.create(targetSubmodelRegistryBaseUrl, mnestixFetch())
+            : undefined;
+
+        return new TransferService(
+            targetAasRepositoryClient,
+            targetSubmodelRepositoryClient,
+            targetAasDiscoveryClient,
+            targetAasRegistryClient,
+            targetSubmodelRegistryClient,
+        );
+    }
+
+    async transferAasWithSubmodels({
```

Comment:
If you don't plan to add unit tests in this substask, please create a new subtask so we don't forget writing tests 😄